### ISSUE TITLE
Failed to load coverage: *.rmeta The file was not recognized as a valid object file

### ISSUE THREAD

**Issue by ktmeaton** [taiki-e/cargo-llvm-cov#342]
Related to #316, when using the `--doc` flag on WSL, the `.rmeta` files are executable and generate the error: 

_" Failed to load coverage: 'target/llvm-cov-target/debug/deps/libmagic-2321c11047278672.rmeta': The file was not recognized as a valid object file"._

Re-using the minimal working example from #316:

```bash
$ git clone --branch v0.15.0 https://github.com/robo9k/rust-magic.git

$ cd rust-magic

$ cargo +nightly --version
cargo 1.77.0-nightly (84976cd69 2024-01-12)

$ cargo +nightly llvm-cov --version
cargo-llvm-cov 0.6.2
```

The following command finishes successfully in WSL (thanks to `.d` files being ignored from #317?):

```bash
$ cargo +nightly llvm-cov

test result: ok. 8 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.02s
```

But the `--doc` flag introduces errors:

```bash
$ cargo +nightly llvm-cov --doc

error: Failed to load coverage: 'target/llvm-cov-target/debug/deps/libmagic-2321c11047278672.rmeta': The file was not recognized as a valid object file
error: failed to generate report: process didn't exit successfully:
```

With the failing process being `llvm-cov report`:

```bash
/home/$USER/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-cov report \
  -instr-profile=target/llvm-cov-target/rust-magic.profdata \
  -object target/llvm-cov-target/debug/deps/libmagic-2321c11047278672.rlib \
  -object target/llvm-cov-target/debug/deps/libmagic-2321c11047278672.rmeta \
  -object target/llvm-cov-target/doctestbins/src_lib_rs_120_0/rust_out \
  -object target/llvm-cov-target/doctestbins/src_lib_rs_35_0/rust_out \
  -object target/llvm-cov-target/doctestbins/src_lib_rs_436_0/rust_out \
  -object target/llvm-cov-target/doctestbins/src_lib_rs_617_0/rust_out \
  -ignore-filename-regex '/rustc/([0-9a-f]+|[0-9]+\.[0-9]+\.[0-9]+)/|^/mnt/c/Users/$USER/Projects/rust\-magic(/.*)?/(tests|examples|benches)/|^/mnt/c/Users/$USER/Projects/rust\-magic/target/llvm\-cov\-target($|/)|^/home/$USER/\.cargo/(registry|git)/|^/home/$USER/\.rustup/toolchains($|/)'
```

If I remove the `rmeta` file as a candidate object, it works perfectly!

```bash
/home/$USER/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-cov report \
  -instr-profile=target/llvm-cov-target/rust-magic.profdata \
  -object target/llvm-cov-target/debug/deps/libmagic-2321c11047278672.rlib \
  -object target/llvm-cov-target/doctestbins/src_lib_rs_120_0/rust_out \
  -object target/llvm-cov-target/doctestbins/src_lib_rs_35_0/rust_out \
  -object target/llvm-cov-target/doctestbins/src_lib_rs_436_0/rust_out \
  -object target/llvm-cov-target/doctestbins/src_lib_rs_617_0/rust_out \
  -ignore-filename-regex '/rustc/([0-9a-f]+|[0-9]+\.[0-9]+\.[0-9]+)/|^/mnt/c/Users/$USER/Projects/rust\-magic(/.*)?/(tests|examples|benches)/|^/mnt/c/Users/$USER/Projects/rust\-magic/target/llvm\-cov\-target($|/)|^/home/$USER/\.cargo/(registry|git)/|^/home/$USER/\.rustup/toolchains($|/)'
```

Could the `.rmeta` files on WSL be ignored as objects, similar to the `.d` files from #317?

**Comment by taiki-e** [comment]
Thanks for the report!
 
> Could the `.rmeta` files on WSL be ignored as objects, similar to the `.d` files from #317?

Yeah, `.rmeta` should be handled that way as well.

**Comment by taiki-e** [comment]
Fixed in 0.6.3 (#343).

**Comment by ktmeaton** [comment]
Works perfectly now, thank you so much!
